app/Foundation/Providers/ModuleServiceProvider.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of Cachet.
+ *
+ * (c) Alt Three Services Limited
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CachetHQ\Cachet\Foundation\Providers;
+
+use Illuminate\View\Compilers\BladeCompiler;
+use Illuminate\Support\ServiceProvider;
+
+class ModuleServiceProvider extends ServiceProvider
+{
+    /**
+     * The module definitions.
+     *
+     * @var array
+     */
+    protected $modules = [
+        'index' => [
+            ['group' => 'messages', 'partial' => 'partials.modules.messages'],
+            ['group' => 'status', 'partial' => 'partials.modules.status'],
+            ['group' => 'components', 'partial' => 'partials.modules.components'],
+            ['group' => 'metrics', 'partial' => 'partials.modules.metrics'],
+            ['group' => 'scheduled', 'partial' => 'partials.modules.scheduled'],
+            ['group' => 'timeline', 'partial' => 'partials.modules.timeline'],
+        ],
+    ];
+
+    /**
+     * The group definitions.
+     *
+     * @var array
+     */
+    protected $groups = [
+        'index' => [
+            'messages'   => 10000,
+            'status'     => 20000,
+            'components' => 30000,
+            'metrics'    => 40000,
+            'scheduled'  => 50000,
+            'timeline'   => 60000,
+        ],
+    ];
+
+    /**
+     * Boot the service provider.
+     *
+     * @param \Illuminate\View\Compilers\BladeCompiler $blade
+     */
+    public function boot(BladeCompiler $blade)
+    {
+        $blade->directive('modules', function ($group = null) {
+            $code = [
+                'foreach (array_numeric_sort($modules) as $group => $parts) {',
```

Comment:
I want this tidied and turned into a single recursive function call of some sort.